src/main/scala/org/broadinstitute/dsde/firecloud/EntityClient.scala

```diff
@@ -248,21 +248,24 @@ class EntityClient (requestContext: RequestContext)(implicit protected val execu
       // 1. The type of the tsv import file (membership, entity, or update)
       // 2. The entity type we're trying to import or update, which is magically verified in the withFoo functions below.
       tsv.firstColumnHeader.split(":") match {
-        case Array( tsvType, entityHeader ) =>
-          val entityType = entityHeader.stripSuffix("_id")
-          if( entityType == entityHeader ) {
+        case h: Array[String] if h.length == 1 || h.length == 2 =>
```

Comment:
suggestion, since `h.head` and `h.last` are a little difficult to read:

```
case entityType :: Nil => //use the default TSV type
case tsvType :: entityType :: Nil => //use the TSV type given
case _ => //error
```

The last line is equivalent to `case tsvType :: theRest`, where `theRest` is an array of at least two, and implies at least one of a) someone's using a colon in an entity name b) they are very confused

Reply:
i'd be inclined to make the result of this `match` block be a `TsvType`. then this function becomes

1. determine the type of the TSV
2. import it

which i'd argue is a touch more readable